src/client/opamSwitchCommand.ml

```diff
@@ -187,7 +187,11 @@ let remove gt ?(confirm = true) switch =
 
 let install_compiler_packages t atoms =
   (* install the compiler packages *)
-  if atoms = [] then t else
+  if atoms = [] then begin
+    OpamFile.Environment.write (OpamPath.Switch.environment t.switch_global.root t.switch) (OpamEnv.compute_updates t);
+    OpamEnv.check_and_print_env_warning t;
+    t
```

Comment:
Won't it be redundant to print the env warning at this point ?

Reply:
There's always the update to `PATH` - you should still run `opam env` even for an empty switch?